Marlin/src/gcode/calibrate/G33.cpp

```diff
@@ -475,6 +479,27 @@ void GcodeSuite::G33() {
 
   if (!_0p_calibration) ac_home();
 
+  #if HAS_DELTA_SENSORLESS_PROBING
+    if (verbose_level > 0 && do_save_offset_adj) {
+      xyz_pos_t reset{0};
+      offset_sensorless_adj = reset;
+
+      float z_at_pt[NPP + 1] = { 0.0f };
+      probe.test_sensitivity = { true, false, false };
+      if (probe_calibration_points(z_at_pt, 1, dcr, false, false, probe_at_offset))
+        probe.save_offset_sensorless(true, z_at_pt[CEN]);
+      z_at_pt[NPP + 1] = { 0.0f };
```

Comment:
warning z_at_pt[NPP + 1]   z_at_pt has size NPP + 1
and I thing z_at_pt is resetted inside probe_calibration_points

Reply:
Perhaps it is not the most orthodox, I do not argue.
The purpose of initializing `z_at_pt[NPP + 1]` is so that `z_at_pt[CEN]` is not saved to each of the three center probes needed to load `offset_sensorless_adj()` (one center probe per axis), none of the three points can be part of deviation statistics for G33.
In fact, I copied line 505, which does the exact same thing at the start of every G33 loop.

Reply:
I don't get the prompt during my builds.
And the same error does not appear in line 505?
There is similar code on line 342, 331. I also don't see any warning during compilation.
Any idea how to fix it, maybe if I declare the float variable every time it is initialized?

Reply:
> inside probe_calibration_points there is this macro that I think set all value to 0 LOOP_CAL_ALL(rad) z_pt[rad] = 0.0f;

Thanks, very good suggestion. I fix it like this.

Reply:
Line 505 is a new variable declaration not a assignment